src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -520,222 +369,161 @@ Let us review the example below where the buffer is allocated for the kernel's f
-           xrtRunSetArg(run,2,scalar_2); // Arguments are specified starting from 0
-           xrtRunStart(run);
-           xrtRunWait(run);
+The ``xrt::ip`` object creation is very similar to creating a kernel. 
 
-           // Close the run handle
-           xrtRunClose(run);
+.. code:: c++
+      :number-lines: 35
+          
+           auto xclbin_uuid = device.load_xclbin("kernel.xclbin");
+           auto ip = xrt::ip(device, xclbin_uuid, "ip_name");
+           
+An ip object can only be opened in exclusive mode. That means at a time, only one thread/process can access IP at the same time. This is required for a safety reason because multiple threads/processes reading/writing to the AXI-Lite registers at the same time potentially leads to a race situation. 
 
-Note the following APIs regarding  the above example
+Allocating buffers for the IP inputs/outputs 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   - The kernel is executed by ``xrtKernelRun`` API by specifying all its arguments to obtain a Run handle
-   - The API ``xrtKernelRun`` is non-blocking. It returns as soon as it submits the job without waiting for the kernel's actual execution start.  
-   - The host code uses ``xrtRunWait`` API to block the current thread and wait till the kernel execution is finished.       
-   - After a run is finished, the same run handle can be reused to execute the kernel multiple times if desired. 
-     
-       - API ``xrtRunSetArg`` is used to set one or more arguments, in the example above only the last (3rd) argument is changed before the second execution
-       - API ``xrtRunStart`` is used to execute the kernel using the run handle. 
-   - API ``xrtRunClose`` is used to close the Run handle.  
- 
-   
-**C++**: The equivalent C++ code
+Simialr to XRT managed kernel ``xrt::bo`` objects are used to create buffers for IP ports. However, the memory bank location must be specified explicitly by providing enumerated index of the memory bank. 
 
-In C++ the ``xrt::kernel`` class provides **overloaded operator ()** to execute the kernel with a comma-separated list of arguments.  
+Below is a example of creating two buffers. Note the last argument of ``xrt::bo`` is the enumated index of the memory bank as seen by the XRT (in this example index 8 corresponds to the host-memory bank). The bank index can be obtained by ``xbutil examine --report memory`` command.  
 
 .. code:: c++
-      :number-lines: 50
-   
-           // 1st kernel execution
-           auto run = kernel(buf_a, buf_b, scalar_1); 
-           run.wait();
-    
-           // 2nd kernel execution with just changing 3rd argument
-           run.set_arg(2,scalar_2); // Arguments are specified starting from 0 
-           run.start();
-           run.wait();
+      :number-lines: 35
+          
+           auto buf_in_a = xrt::bo(device, DATA_SIZE, xrt::bo::flags::host_only, 8);
+           auto buf_in_b = xrt::bo(device, DATA_SIZE, xrt::bo::flags::host_only, 8);
```

Comment:
This is a general comment, not overly important.  While I like the types on the right, and while much of C++ makes this natural, object construction is different.  It really should be
```
xrt::bo buf_in_a(device, DATA_SIZE, xrt::bo::flags::host_only, 8);
xrt::bo buf_in_b(device, DATA_SIZE, ...);
```
Don't care if we change or leave as is.

Reply:
got it.. I will change this later, I generally like this style so it is all over the place now. But I can change it later if for constructor this is not a norm.

Reply:
I do not think the object construction is different nowadays, with the copy elision.
So it is just a question of taste.